src/critics/reference_trajectory_critic.cpp

```diff
@@ -13,48 +13,36 @@ void ReferenceTrajectoryCritic::initialize()
   getParam(reference_cost_power_, "reference_cost_power", 1);
   getParam(reference_cost_weight_, "reference_cost_weight", 3.0);
 
-  getParam(enable_nearest_path_angle_critic_, "enable_nearest_path_angle_critic", true);
```

Comment:
So I definitely cannot argue with decoupling the reference trajectory critic from other critics, but how are we able to get the same performance without these 2 additional critics internally? I thought before that's the only way we could get the performance that were happy with w.r.t. path following accurately (and in the right direction)

Reply:
Are you intending that both of these are still used together for the most part?

Reply:
Maybe rename would be sufficient

Reply:
I don't understand that resolution. Are these 2 critics really meant to both be used at the same time? Is there a situation we'd want to use the new `local goal` critic in the absence of this one? If not, then we should just keep them combined.

Reply:
I answered more detailed somewhere. 
I think we should decouple them because they have different purpose, and potentially can be used separately, but for now i see better behavior when they work together. reference critic from my point of view can be renamed to path align critic

Reply:
OK, assuming they're renamed to be appropriate to what they do, I can accept that. I would though say that both techniques are a little hand-tuned and based on the reference input path, so keeping all that in 1 place can be nice and just accept that our path following critic is a little bespoke.

Reply:
renamed ref critic to path_align_critic